### PR TITLE
docs: add adr 003-unit-tests-folder-structure

### PR DESCRIPTION
**Pull request by aaschlote**

### CODE REVIEW

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Accepted
+
+## Context
+
+Our team acknowledges several issues with our current unit test folder structure (`\tests\unit\`):
+
+- Difficulty in locating unit tests corresponding to implementation code
+- Lack of clarity regarding test coverage
+- Challenges in writing new unit tests for newly implemented code due to separation from the codebase.
```
Comment (frederike-ramin) [inline doc/adr/0003-unit-tests-folder-structure.md:15]: Maybe add:
- need to independently rearrange test folders when files move folders

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+To address these concerns, we have identified the following options:
+
+1. Leave as is
```
Comment (frederike-ramin) [inline doc/adr/0003-unit-tests-folder-structure.md:19]: maybe `and accept the drawbacks`?

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+2. Add `__test__` folder within lowest-level app folders
+3. In-source tests
+4. Parallel test files in the same folder as app files
```
Comment (frederike-ramin) [inline doc/adr/0003-unit-tests-folder-structure.md:22]: Can you switch 3 and 4? I feel like the tests move closer to the code with the options, so having in-source as last would make sense?

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+The pros and cons of these options are outlined in the image below:
+
+![Unit test folder structure options](/doc/images/unit_tests_folder_structure.jpg)
```
Comment (frederike-ramin) [inline doc/adr/0003-unit-tests-folder-structure.md:26]: Can you rename the options in the image to be exactly the same as outlined above?

**Review by frederike-ramin — COMMENTED** [review]
Nice. :)

File: doc/adr/0003-unit-tests-folder-structure.md (hunk 3 above)
Comment (aaschlote) [reply doc/adr/0003-unit-tests-folder-structure.md]: It sounds reasonable for me :)

File: .prettierignore [hunk 5]
```diff
@@ -21,3 +21,4 @@ Dockerfile
 *.pdf
 *.generated.ts
 sonar-project.properties
+doc/images/
```
Comment (chohner) [inline .prettierignore:24]: missing newline 🙈

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+## Status
+
+Accepted
+
+## Context
+
+Our team acknowledges several issues with our current unit test folder structure (`\tests\unit\`):
+
+- Difficulty in locating unit tests corresponding to implementation code
+- Lack of clarity regarding test coverage
```
Comment (chohner) [inline doc/adr/0003-unit-tests-folder-structure.md:14]: I feel like this point isn't solved by test location but through a coverage tool 🤔

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+- Challenges in writing new unit tests for newly implemented code due to separation from the codebase.
```
Comment (chohner) [inline doc/adr/0003-unit-tests-folder-structure.md:15]: Whats the challenge here? Maybe `Easy to forget adding unit tests for new code`?

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+## Decision
+
+As a group we decided for the option 2 (`Add __test__ folder within lowest-level app folders`)
+
+- This option requires minimal configuration and is easily implemented.
+- Leveraging our current unit testing framework, `jest`, which already recognizes test files within `__test__` folders.
+- Offers a more intuitive location for unit tests, particularly in the TypeScript and JavaScript ecosystem.
+
+While the team is content with this decision, we acknowledge the potential consideration of option 3 for long-term planning.
```
Comment (chohner) [inline doc/adr/0003-unit-tests-folder-structure.md:37]: This is now option 4, right?

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+To address these concerns, we have identified the following options:
+
+1. Leave as is and accept the drawbacks
+2. Add `__test__` folder within lowest-level app folders
+3. Parallel test files in the same folder as app files
+4. In-source tests
+
+The pros and cons of these options are outlined in the image below:
+
+![Unit test folder structure options](/doc/images/unit_tests_folder_structure.jpg)
```
Comment (chohner) [inline doc/adr/0003-unit-tests-folder-structure.md:27]: I would even go as far as to just copy the text content to this doc (I know, annoying), but images are not searchable nor adaptable in the future

File: doc/adr/0003-unit-tests-folder-structure.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+- This option requires minimal configuration and is easily implemented.
+- Leveraging our current unit testing framework, `jest`, which already recognizes test files within `__test__` folders.
+- Offers a more intuitive location for unit tests, particularly in the TypeScript and JavaScript ecosystem.
+
+While the team is content with this decision, we acknowledge the potential consideration of option 3 for long-term planning.
+
+## Consequences
+
+- Incorrect configuration may inadvertently include test code in the production bundle.
```
Comment (chohner) [inline doc/adr/0003-unit-tests-folder-structure.md:41]: I thought the consequence is adding a ticket to the backlog? the bullet point should be an AC of that ticket

File: doc/adr/0003-unit-tests-folder-structure.md (hunk 6 above)
Comment (aaschlote) [reply doc/adr/0003-unit-tests-folder-structure.md:14]: True, this point is not related what we are trying to change it

Comment (aaschlote) [reply doc/adr/0003-unit-tests-folder-structure.md:14]: I'll remove it

File: doc/adr/0003-unit-tests-folder-structure.md (hunk 7 above)
Comment (aaschlote) [reply doc/adr/0003-unit-tests-folder-structure.md:15]: It has been a challenge sometimes to write new unit tests for new code. I guess the terms ``Challenges`` and ``Easy to forget`` are the same in this sentence. I will keep like this :)

File: doc/adr/0003-unit-tests-folder-structure.md (hunk 10 above)
Comment (aaschlote) [reply doc/adr/0003-unit-tests-folder-structure.md:41]: it's one of the consequence, but the bigger one is to add the test code in the production bundle, but of course this should an AC in the ticket

**Review by chohner — APPROVED** [review]
I still think `Challenges in writing new unit` might be a bit too vague in the future but not a blocker!

**Review by pgurusinga — APPROVED** [review]

**Pull request merged**